wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/spi/environment/ResourceLimitAttributesProvisioner.java

```diff
@@ -58,10 +58,10 @@ public static void provisionMemory(
           "Requested default container resource limit is less than default request. Request parameter will be ignored.");
     }
 
-    if (memoryLimit <= 0) { // if limit only is undefined
+    if (memoryLimit <= 0 && defaultMemoryLimit > 0) { // if limit only is undefined
```

Comment:
I know that this was here before, but "if limit only is undefined" requires rephrasing :)
Also what should happen if the default memory limit is precisely 0?

Reply:
the same as -1